src/main/java/com/coveros/selenified/element/check/State.java

```diff
@@ -0,0 +1,314 @@
+    String NOTPRESENT = " is not present on the page";
+    String DISPLAYED = " is displayed on the page";
+    String NOTDISPLAYED = " is not displayed on the page";
+    String CHECKED = " is checked on the page";
+    String NOTCHECKED = " is not checked on the page";
+    String EDITABLE = " is editable on the page";
+    String NOTEDITABLE = " is not editable on the page";
+    String ENABLED = " is enabled on the page";
+    String NOTENABLED = " is not enabled on the page";
+
+    // ///////////////////////////////////////
+    // assessing functionality
+    // ///////////////////////////////////////
+
+    /**
+     * Verifies that the element is present. If the element isn't present, it
+     * waits up to the default time (5 seconds) for the element, before marking
+     * this verification as a failure. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     */
+    void present();
+
+    default boolean checkPresent(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + PRESENT, waitFor);
+        // perform the check
+        boolean isPresent = getElement().is().present();
+        // record the result
+        if (isPresent) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + PRESENT, timeTook, Success.PASS);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTPRESENT, timeTook, Success.FAIL);
+        }
+        return isPresent;
+    }
+
+    /**
+     * Verifies that the element is not present. If the element is present, it
+     * waits up to the default time (5 seconds) for the element to be removed,
+     * before marking this verification as a failure. This information will be
+     * logged and recorded, with a screenshot for traceability and added
+     * debugging support.
+     */
+    void notPresent();
+
+    default boolean checkNotPresent(double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + NOTPRESENT, waitFor);
+        // perform the check
+        boolean isPresent = getElement().is().present();
+        // record the result
+        if (isPresent) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + PRESENT, timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + NOTPRESENT, timeTook, Success.PASS);
+        }
+        return !isPresent;
+    }
+
+    /**
+     * Verifies that the element is displayed. If the element isn't present, it
+     * waits up to the default time (5 seconds) for the element, before marking
+     * this verification as a failure. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     */
+    void displayed();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "displayed" to prevent any misunderstanding/clash with field "DISPLAYED" defined on line 42. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)